Clarify esm restrictions (#13633)
getsentry/sentry-docs: docs/platforms/javascript/common/install/esm-without-import.mdx

When running your application in ESM mode, you will most likely want to <PlatformLink to="/install/esm">follow the ESM instructions</PlatformLink>. However, if you want to avoid using the `--import` command line option, for example if you have no way of configuring a CLI flag, you can also follow an alternative setup that involves importing the `instrument.mjs` file directly in your application.

<Alert level='warning' title='Restrictions of this installation method'>

> This installation method has the fundamental restriction that only native Node.js APIs can be instrumented (like for example `fetch` and the `http` module).

Reviewer: ```suggestion
This installation method has the fundamental restriction that only native Node.js APIs can be instrumented (such as `fetch` and the `http` module).
```

> In other words, this means that the Sentry SDK will not capture data for database calls, queues, ORMs, libraries or other framework-specific data.

Reviewer: ```suggestion

As a result, the Sentry SDK will not capture data from database calls, queues, ORMs, third-party libraries, or other framework-specific data.
```
